### PR TITLE
switch to multidistro travis

### PR DESCRIPTION
**Pull request by avsm**

### CODE REVIEW

File: .travis-ci.sh [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
 git clone git://github.com/mirage/mirage-www
 cd mirage-www
 
-opam install mirage
-make MODE=$MIRAGE_MODE configure
-make MODE=$MIRAGE_MODE build
+for mode in unix xen; do
+  export OPAMYES=1
+  eval `opam config env`
+  make MODE=$MIRAGE_MODE configure
```
Comment (hannesm) [inline .travis-ci.sh]: should be `$mode` (what we are looping over) here...

**Comment by samoht** [comment]
Did we loose OSX testing?

**Comment by avsm** [comment]
was it ever active on this repo?

**Comment by samoht** [comment]
I'm not sure :-) I am just wondering if the new multi-distro scipts (which are awesome btw) are also compatible with OSX. But that's maybe not the place for this discussion :-)

**Comment by hannesm** [comment]
4.01 support is no longer in tcp/ip... and one real [faliure](https://s3.amazonaws.com/archive.travis-ci.org/jobs/117154871/log.txt):
```
make[1]: Leaving directory `/repo/mirage-www/stats'
mirage configure -f src/config.ml  -t xen
Error: Nothing to do
opam-depext: internal error, uncaught exception:
             Failure("OS package installation failed")
             
[ERROR]
  The command "opam depext c3 cow cowabloga crunch cstruct functoria io-page lwt mirage-bootvar-xen mirage-clock-xen mirage-conduit mirage-console mirage-entropy-xen mirage-http mirage-net-xen mirage-types mirage-types-lwt mirage-types>=2.6.0 mirage-xen nocrypto tcpip xapi-rrd xen-evtchn xen-gnt xenstore zarith-xen" exited with code 1.
make: *** [configure] Error 1
```

**Comment by djs55** [comment]
On CentOS 6 and 7 there's a failure installing `xen-devel`:
```
    yum install -y xen-devel
Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: centos.netnitco.net
 * extras: chicago.gaminghost.co
 * updates: centos.chi.host-engine.com
No package xen-devel available.
Error: Nothing to do
OS package installation failed
[ERROR]
```
I believe `xen-devel` is not supported in RHEL, so isn't in base CentOS. People who really want to use Xen in CentOS have to first [enable the Virtualisation Sig extra repo](https://wiki.centos.org/HowTos/Xen/Xen4QuickStart) and try again. The commands you would need are:
```
yum install centos-release-xen
yum update
yum install xen
```
Unfortunately because `yum install centos-release-xen` installs an extra yum repo and gpg key on your system, it's not possible to combine these lines together and say `yum install centos-release-xen xen-devel` because it won't find `xen-devel` until after it has installed `centos-release-xen`. So it's not very `opam depext` friendly, although we could write a Dockerfile for it?

It might be easiest to drop or otherwise ignore CentOS for now until we have a nice solution for this.

**Comment by hannesm** [comment]
could you please rebase instead of merge?  the diff of this PR is way too long for what it is supposed to do

**Comment by avsm** [comment]
I'll rebase once this goes green.

**Comment by yomimono** [comment]
I'm closing this as it's very unmergeable; please submit a new rebased PR if you're still interested in using the multidistro travis scripts to test tcpip.

**Pull request closed without merging**